<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="AY" />


<title>Automated workflow for ChIP-Seq analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">chipseq snakemake</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="contact.html">
    <span class="fa fa-envelope fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://github.com/yeroslaviz">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/AssaYeroslaviz">
    <span class="fa fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://www.linkedin.com/in/ayeroslaviz/">
    <span class="fa fa-linkedin fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Automated workflow for ChIP-Seq analysis</h1>
<h4 class="author">AY</h4>
<h4 class="date">2/10/2020</h4>

</div>


<p>This workflow should able an automated workflow of chip-seq data sets.</p>
<p>An <a href="environment.yaml">environemnt.yaml</a> file was created to contain the list of packages installed for the environment.</p>
<p>The installation with the above file failed due to conflicts, so I decided to do it manually. I have created the conda env with the command <code>conda create -n ChipSeq.Snakemake</code>. Then, after activating the environment, I installed the tools manually.</p>
<ul>
<li>deeptools</li>
<li>meme, macs2</li>
<li>samtools segemehl ucsc-wigtobigwig trimmomatic fastqc snakemake-minimal</li>
<li>bwa</li>
<li>bowtie2</li>
</ul>
<p>(e.g. <code>conda install bwa</code>)</p>
<p>A [config.yaml] file contains all the needed parameters for the analysis. For each analysis, some parameters <strong>must</strong> be changed for the workflow to run correctly.</p>
<ol style="list-style-type: decimal">
<li>link to the organism’s fastA file</li>
<li>link to the gtf file</li>
</ol>
<div id="creating-the-conda-environment" class="section level1">
<h1>creating the conda environment</h1>
<pre class="bash"><code>conda create --name chipseq

conda activate chipseq

conda config --add channels bioconda # add the bioconda channel to the conda env.

conda install snakemake-minimal  samtools bedtools cutadapt fastqc rseqc bwa bowtie2 picard macs2 bioconductor-chipseeker segemehl deeptools meme segemehl -c bioconda -c conda-forge -c r -c bioinfo
conda install -c conda-forge graphviz # to enable the dot tool to create directed graphs</code></pre>
</div>
<div id="setting-the-config.yaml-file" class="section level1">
<h1>setting the <code>config.yaml</code> file</h1>
<p>the same config file used for the genome indexing and star mapping is also used here.</p>
<p>This file contains a list of possible organisms with links to their <code>fastA</code> and <code>gtf</code> file. These links were used in the the snakefile genomeIndexing to create the indicies for the different mappers.</p>
<p>For now, the mappers we have are star (for RNA-Seq usually) and bwa, segemehl, bowtie2 for chip-seq data sets. General parameters for each of these mappers can be eset in the config file</p>
<p>One parameter set here is which mapper to use for the analysis.</p>
<pre class="bash"><code>mapper: &quot;bowtie2&quot;</code></pre>
<p>Here the value can be one of the three above mentioned mappers <code>bowtie2</code>,<code>bwa</code>, or <code>segemehl</code>.</p>
<pre class="bash"><code># For the bedgraph files.
effGenomeSize: 2652783500</code></pre>
<p>The effektive genome size parameter is set for deepTools to calculate the values in each bin in <code>bamCoverage</code>.</p>
</div>
<div id="the-snakefile" class="section level1">
<h1>the <code>snakefile</code></h1>
<p>For the workflow a <a href="ChipSeq.Snakemake"><code>snakefile</code></a> was created. Here, each steps is described as a rule and they are executed in a directed order of steps, each one depending on one (or few) steps before. The order of execution is also visualized in the attached image.</p>
<p><img src="figures/dag.chipseq.workflow.snakemake.svg" /></p>
<p>It begins with the fastq files, converts them to different formats (bdg, bam, bai, peaks, etc.) and run different analyses.</p>
<p># Snakefile run</p>
<p>the snakefile run starts with mapping the files using the given mapper (in the example here I’ll use <code>bowtie2</code>) (the samples in the examples commands were chosen arbitrary.)</p>
<pre class="bash"><code>bowtie2 --threads 16 --dovetail --very-sensitive-local --no-unal --no-mixed --no-discordant -X 2000 -x /fs/pool/pool-bcfngs/genomes/Dme.BDGP6.22/bowtie2Index/Dme.BDGP6.22 -1 2Aub_W_2_22_11.conc.R1.fastq.gz -2 2Aub_W_2_22_11.conc.R2.fastq.gz 2&gt; P000/Dme.BDGP6.22/bowtie2/2Aub_W_2_22_11.log | samtools view -Sbhu -q 7 -@8 - | samtools sort -@8 - -o P000/Dme.BDGP6.22/bowtie2/2Aub_W_2_22_11.sorted.bam

picard MarkDuplicates I=P000/Dme.BDGP6.22/bowtie2/in_W_2_22_11.sorted.bam O=P000/Dme.BDGP6.22/bowtie2/in_W_2_22_11.rmdup.bam M=P000/Dme.BDGP6.22/bowtie2/in_W_2_22_11_rmdup_metric.txt REMOVE_DUPLICATES=true

samtools index P000/Dme.BDGP6.22/bowtie2/2Aub_W_2_22_11.rmdup.bam</code></pre>
<p>Next the <code>bam</code> files are converted to <code>bedgraph</code></p>
<pre class="bash"><code>bamCoverage --bam P000/Dme.BDGP6.22/bowtie2/2Aub_W_2_22_11.rmdup.bam --outFileName P000/Dme.BDGP6.22/bowtie2/bedgraph/2Aub_W_2_22_11.rmdup.bdg --outFileFormat bedgraph --binSize 10 --effectiveGenomeSize 2652783500 --extendReads 160 --normalizeUsing CPM --ignoreForNormalization chrX chrM</code></pre>
<p>Peak calling is done using the <code>macs2</code> tool</p>
<pre class="bash"><code>macs2 callpeak -t P000/Dme.BDGP6.22/bowtie2/2Aub_W_2_22_11.rmdup.bam -c P000/Dme.BDGP6.22/bowtie2/in_W_2_22_11.rmdup.bam -f BAM -g mm -n W_2_22_11 -B -q 0.05 --outdir P000/Dme.BDGP6.22/bowtie2/macs2/
macs2 callpeak -t P000/Dme.BDGP6.22/bowtie2/2Aub_W_2_22_11.rmdup.bam -c P000/Dme.BDGP6.22/bowtie2/in_W_2_22_11.rmdup.bam -f BAM -g mm -n W_2_22_11 --broad -B -q 0.05 --outdir P000/Dme.BDGP6.22/bowtie2/macs2/</code></pre>
<p>And the bedgraph files can also be comapred with <code>macs2</code> to generate fold-enrichment track.</p>
<pre class="bash"><code>macs2 bdgcmp -t P000/Dme.BDGP6.22/bowtie2/bedgraph/2Aub_W_2_22_11.rmdup.bdg -c P000/Dme.BDGP6.22/bowtie2/bedgraph/in_W_2_22_11.rmdup.bdg -o W_2_22_11 --outdir P000/Dme.BDGP6.22/bowtie2/macs2/W_2_22_11.compared.bdg -m FE</code></pre>
<p><strong>add detailed-summary</strong> when the workflow is done</p>
<p><strong>chipseeker</strong></p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
