---
title: "Automated workflow for ChIP-Seq analysis"
author: "AY"
date: "2/10/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

This workflow should able an automated workflow of chip-seq data sets.


An [environemnt.yaml](environment.yaml) file was created to contain the list of packages installed for the environment.

The installation with the above file failed due to conflicts, so I decided to do it manually. I have created the conda env with the command `conda create -n ChipSeq.Snakemake`. Then, after activating the environment, I installed the tools manually.

- deeptools
- meme, macs2
- samtools segemehl ucsc-wigtobigwig trimmomatic fastqc snakemake-minimal 
- bwa 
- bowtie2

(e.g. `conda install bwa`)

A [config.yaml] file contains all the needed parameters for the analysis. For each analysis, some parameters __must__ be changed for the workflow to run correctly.

1. link to the organism's fastA file
2. link to the gtf file

# creating teh conda environment

```bash 
conda env create --name chipseq --file environment.yml
```






